Add tracing sections for gateway 3.0

mayocream opened the pull request:
### Summary
<!-- Description of PR, with any special instructions for your reviewers. -->
Add observability sections for tracing, log and metrics.

### Reason
<!-- Why are you making this change? Can be a link to a Jira ticket, GH issue, 
Trello card, etc. -->

[FT-2922](https://konghq.atlassian.net/browse/FT-2922)

Guaris commented on app/gateway/3.0.x/observability/logs.md:
```diff
@@ -0,0 +1,125 @@
+Log levels are set in [Kong's configuration](/gateway/{{page.kong_version}}/reference/configuration/#log_level). Following are the log levels in increasing order of their severity: `debug`, `info`,
+`notice`, `warn`, `error` and `crit`.
+
+- *`debug`:* It provides debug information about the plugin's runloop and each individual plugin or other components. Only to be used during debugging since it is too chatty.
```
```suggestion
- *`debug`:* It provides debug information about the plugin's run loop and each individual plugin or other components. Only to be used during debugging since it is too chatty.
```

mayocream commented:
> Is this still a draft @mayocream ?

It's ready for review now!

Guaris reviewed (commented):
@mayocream What is `telemtry.md` supposed to be? It feels incomplete.

mayocream commented:
> @mayocream I reviewed this. I'm not sure what the purpose of `telemetry.md` is. It feels incomplete.
> 
> I also rebased and moved the docs into the new navigation bar.

I removed the `telemetry.md`, the purpose was to tell the 3rd developers that it's recommended to use the new tracing API to monitor and track the performance and events of the custom plugin.

mayocream commented:
@Guaris Thanks for the copy edit; the observability/tracing sections should be under "Kong in Production" nav in my own perspective since it's a common toolchain like the [perf testing framework](https://deploy-preview-4014--kongdocs.netlify.app/gateway/latest/understanding-kong/kong-performance-testing/).

Guaris commented:
LGTM
